Nerfs ID cards

comma opened the pull request:
They no longer show full name and job in their name, moved that to examine bit instead.
PDAs got same treatment.

In exchange, they get a linkie when you examine a guy, letting you see the ID (in full detail with pic and all if you're adjacent)

IMO should make stowaways/impersonations less obvious while still letting legit seeing who is who.

<!-- 
Do not forget to add a changelog when you have made admin/player facing changes that can alter gameplay.
Examples which require a changelog entry include:
* Adding/removing objects that players may interact with, or the way they function.
* Adding/removing/altering admin tools.
* Changing the map.

Examples were changelog entries are optional/not typically required:
* Cosmetic changes such as descriptions, sound effects, etc.
* Optimizations and other changes to underlying systems which do not affect gameplay.
* Minor bug fixes.

You find a README and example file in .\html\changelogs\ for further instructions.
-->

comma committed:
Nerfs ID cards

They no longer show full name and job in their name, moved that to examine bit instead.
PDAs got same treatment.

In exchange, they get a linkie when you examine a guy, letting you see the ID (in full detail with pic and all if you're adjacent)

IMO should make stowaways/impersonations less obvious while still letting legit seeing who is who.

Casper3667 commented:
I wonder.. Does this do anything against people with masks appearing as Unknown (Urist McGasmask)?

comma commented:
Nope, doesn't touch that. For the record both legit and non legit people appear that (i.e.regardless of if ID actually matches real name)

atlantiscze commented:
Do we have to serve /**everything**/ to antags on a diamond tipped golden platter?

Being able to tell what job the person has by glancing on their ID is a good thing. Do not remove that.

ghost commented:
Sounds like it'll make things interesting. :+1:
EDIT: Can this be done with badges, too? I desperately need an excuse to shove my credentials in people's faces.

comma commented:
Badges are bit trickier since they're not shown on their own on examine.
I think I'll add some sort of BADGE NUMBER for enhanced cop RP and some immediate identification
Also need to figure out why these holobadges don't get name and job added to examine ree

comma committed:
Makes badges work like that too

They are high visibility now so you don't have to click a link to click a link to read them.
Added BADGE NUMBERS to holobadges for enhanced cop RP

comma commented:
![](https://i.imgur.com/EOklCel.png)

Sbotkin commented:
MFW this PR makes ID card not ID.

Techhead0 commented on code/modules/clothing/under/accessories/badges.dm:
```diff
@@ -70,6 +91,16 @@
 
+/obj/item/clothing/accessory/badge/holo/set_name(var/new_name)
+	..()
+	badge_number = random_id(type,100,999)
```
Three digit IDs are really ubiquitous ingame and I think this is something where something like a 5 digit ID might feel better.

comma replied:
I just went with 3 because easily readable and 5 might be too much for average player's brain capacity. But yeah, it did remind me of monkies and such, gonna bump it to 4

Dukica99 commented:
This would be fine if examine expansions (pressing on that text to expand more info) weren't so laggy/prone to freezing. :(

comma commented:
I don't know what to tell you, href links is how pretty much every UI works in game.

FreePrivateer commented:
Please no. I really don't want to have to open the crew manifest and cross reference someone's name to their job every time I see someone (because there's no continuity and they could be playing anything from off-duty to some other department even if I /do/ recognize them). I get the intention behind the change, and it'd be a cool idea if the game worked like real life where we generally recognize our co-workers, or when everyone wore departmental rank and uniform. But with lack of real EC uniform regs, civvies running around everywhere, and the difficulty of seeing position in a uniform environment that gives you a few pixels to work on, removing name and rank off the IDs just seems like a effort in misery.

Maybe have the ID at /least/ show department and rank or something (An enlisted deck department ID)? Its a bit unrealistic considering off-duty people aren't going to be swapping their IDs out, but it seems the only easy compromise.

comma commented:
>I really don't want to have to open the crew manifest and cross reference someone's name to their job every time I see someone (because there's no continuity and they could be playing anything from off-duty to some other department even if I /do/ recognize them).

Then don't. Examine. Click link.
Does no one actually read the description of PR before complaining. Hell I even got a screenshot.

FreePrivateer commented:
Ah, apologies. I read that initially to mean you would only get the examine verb if you were adjacent. I still think it would be useful to have enlisted/civilian/officer department on the ID, but that's more workable.

comma commented:
crew/passngers get different IDs, and for enlisted-officer you're better off looking at ranks because of weird jobs that can be both.

BlueNexus commented:
So, what do we do if the person doesn't have their ID in their PDA (like a heathen) and we want to figure out their job?

comma commented:
https://github.com/Baystation12/Baystation12/pull/21318#issuecomment-383835661
I'm going to ignore further questions that can be answered by /reading/

sabiram commented on code/modules/clothing/under/accessories/badges.dm:
```diff
@@ -70,6 +91,16 @@
+/obj/item/clothing/accessory/badge/holo/examine(user)
+	..()
+	if(badge_number)
+		to_chat(user,"Badge number is [badge_number].")
```
The badge number is

babydoll replied:
The badge number is...

babydoll commented on code/modules/clothing/under/accessories/badges.dm:
```diff
@@ -36,6 +55,7 @@
 /obj/item/clothing/accessory/badge/attack(mob/living/carbon/human/M, mob/living/user)
 	if(isliving(user))
 		user.visible_message("<span class='danger'>[user] invades [M]'s personal space, thrusting \the [src] into their face insistently.</span>","<span class='danger'>You invade [M]'s personal space, thrusting \the [src] into their face insistently.</span>")
+		to_chat(M, "<span class='warning'>It reads: [stored_name], [badge_string].</span>")
```
Not checking if these are set
